### DISCUSSION TITLE
Formatter - Preserve new lines

### DISCUSSION THREAD

**Discussion by jameskennard** [biomejs/biome#4054] [Ideas]
Would it be feasible to add configuration to the formatter to allow new lines to be preserved?

For example:
```
personBuilder
    .withName('jane Doe')
    .withPhoneNumber('00000 000 000')
    .build()
```
Gets formatted as 
```
personBuilder.withName('jane Doe').withPhoneNumber('00000 000 000').build()
```

In my opinion, this is less readable, and as a developer I may have explicitly chosen to add new lines. Is there any chance of adding the ability to preserve new lines?

Thanks

James.

**Comment by Atulin** [comment]
+1 to that. Alternatively, a setting to automatically split a chain into multiple lines, perhaps with a setting configuring how many functions would need to be chained for that. For example, with some

    "splitLongFunctionChains": 2

code like `a.b().c()` would not be split, but `a.b().c().d()` would be split into multiple lines.
